_posts/2021-03-14-Julia-1.6-what-has-changed-since-1.0.md

```diff
@@ -0,0 +1,737 @@
+There have been too many cool new things (as this post will detail) to stay back to only 1.0 features.
+Already a lot of packages have dropped support for Julia versions older than 1.3.
+
+This post is kind of a follow-up to my [Julia 1.0 release run-down]({{site.url}}//2018/06/01/Julia-Favourite-New-Things.html).
+But it's going to be even longer, as it is covering the last 5 releases since then; and I am not skipping the major new features.
+I am writing this not to break down release by release,
+but to highlight features that had you only used Julia 1.0, you wouldn't have seen.
+Full details can be found in the [NEWS.md](https://github.com/JuliaLang/julia/blob/backports-release-1.6/NEWS.md), and [HISTORY.md](https://github.com/JuliaLang/julia/blob/backports-release-1.6/HISTORY.md)
+
+
+## Front-end changes
+### Soft-scope in the REPL
+Julia 1.0 removed the notion of soft-scope from the language.
+I was very blasé about the change to for-loop bindings in my [1.0 release post](https://www.oxinabox.net/2018/06/01/Julia-Favourite-New-Things.html#for-loop-variable-binding-changes).
+In fact, I didn't even mention this particular change.
+It was [#19324](https://github.com/JuliaLang/julia/pull/19324) for reference.
+
+This was undone in in Julia 1.5 with [#33864](https://github.com/JuliaLang/julia/pull/33864) for in the REPL only.
+Now in the REPL, assigning to a global variable within a for-loop actually assigns that variable, rather than shadowing it with a new variable in that scope.
+The same behavior outside the REPL now gives a warning.
+
+Personally, this change never affected me because I never write for-loops that assign variables at global scope.
+Indeed basically all I code I write is inside functions.
+But I do see how this causes problems for some interactive workflows, especially e.g. when demonstrating something.
+See the [main GitHub issue](https://github.com/JuliaLang/julia/issues/28789) and the longest [Discourse thread](https://discourse.julialang.org/t/another-possible-solution-to-the-global-scope-debacle/15894), though there were many others.
+It took over a year of discussion to work out the solution, particularly because many of the more obvious solutions would be breaking in significant ways.
+
+### Deprecations are muted by default
+
+Julia 1.5+ doesn't display deprecation warnings, unless you run julia with the `--depwarn=yes` flag.
+This was me in [this PR](https://github.com/JuliaLang/julia/pull/35362).
+It's not something I am super happy about.
+Though I think it makes sense.
+Using deprecated methods is actually fine, as long as your dependencies follow SemVer, and you use `^` (i.e. default) bounding in your Project.toml's `[compat]`, which everyone does, because it's enforced by the auto-merge script in the General registry.
+Often it is even necessary if you want to keep compatibility for a while.
+
+Solving deprecations is a thing you should actively chose to do.
+Not casually, when trying to do something else.
```

Comment:
Broken clause - try writing the full sentence.
"Solving dependencies should not be done casually; In particular ..."

Reply:
I actually think "Solving deprecations ... do, not casually ... else." should be one sentence and "In particular ... " be a separate sentence.